src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala

```diff
@@ -109,40 +109,38 @@ class HouseholdFleetManager(
         for {
           neededVehicleCategory              <- requireVehicleCategoryAvailable
           emergencyHouseholdVehicleGenerator <- maybeEmergencyHouseholdVehicleGenerator
-          vehicle                            <- emergencyHouseholdVehicleGenerator.createVehicle(personId, nextVehicleIndex, neededVehicleCategory)
+          vehicle <- emergencyHouseholdVehicleGenerator.createVehicle(
```

Comment:
Should we push this the other way? As in do not extract the generator and use it unless the availableVehicles is empty.